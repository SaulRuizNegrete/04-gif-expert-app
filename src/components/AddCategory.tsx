import { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }: any) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }: any) => {
        setInputValue( target.value )
    }

    const onSubmit = ( event: any ) => {
        event?.preventDefault()
        if( inputValue.trim().length <= 1) return
        //setCategories( (categories: any) => [...categories, inputValue])
        onNewCategory( inputValue.trim() )
        setInputValue('')
    }

    return (
        <form onSubmit={ (event) => onSubmit(event) } aria-label='form'>
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ (event) => onInputChange(event) }
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}