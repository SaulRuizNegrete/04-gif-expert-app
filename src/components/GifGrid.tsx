

export const GifGrid = ({ category }: any) => {
    return (
        <>
            <h3>{ category }</h3>
        </>
    )
}