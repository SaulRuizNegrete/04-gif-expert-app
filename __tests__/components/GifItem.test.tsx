import React from 'react'
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/index'

describe('first', () => { 

    const title = "Mi Título"
    const url = "https://one-punch.com/saitama.jpg"

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar la imagen con la URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url}/>)
        //expect( screen.getByRole('img').src).toBe(url)

         const {src , alt } = screen.getByRole('img') as HTMLImageElement;
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe de mostrar el título en el componente', () => { 
        render(<GifItem title={title} url={url}/>)
        expect( screen.getByText(title) ).toBeTruthy()
     })
 })
