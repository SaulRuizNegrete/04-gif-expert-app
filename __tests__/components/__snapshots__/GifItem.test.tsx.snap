// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mi Título"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Mi Título
    </p>
  </div>
</div>
`;
